crates/vault-registry/src/types.rs

```diff
@@ -185,6 +187,159 @@ pub mod liquidation_vault_fix {
     }
 }
 
+pub mod v4 {
```

Comment:
I'm not sure how exactly the migrations work. Have @sander2 or @gregdhill checked this code?

Reply:
Nope, maybe then hold off on merging until one of them has had a chance to review next week.